init.sls

```diff
@@ -72,20 +72,31 @@ chown_pgdata:
 
 # Deploy users. Sort them, so order is not changing between salt runs
 {% for config in pillar['postgresql']['users']|default({}) %}
-createuser-{{ loop.index }}:
+createuser-{{ config['username'] }}:
```

Comment:
Why did you change this? `loop.index` was used here to allow multiple rules per user.

Reply:
I changed this to have a well defined dependency mapping in the database creation state. I am not quite sure, what you mean by "multiple roles per user", as the role is basically the user.

Reply:
Sorry, I mean rules, not roles.

You need to be able to set multiple rules for a single user, like e.g. allowing two machines for replication acces:

```yaml
    - username: replicant
      type: host
      database: replication
      address: 1.2.3.4/32
      method: trust
      options: NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN REPLICATION
    - username: replicant
      type: host
      database: replication
      address: 1.2.3.5/32
      method: trust
      options: NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN REPLICATION
```

In your approach, this would result in two items with the identifier `username`, and therefore salt would complain.

Reply:
I see. Yes this is lacking atm. This is what I meant with the initial comment, that the hba creaton needs some more care.

Reply:
Well, this is actually already implemented, just don't remove the `loop.index` :)